src/main/java/org/kiwiproject/jersey/client/ServiceIdentifier.java

```diff
@@ -0,0 +1,129 @@
+import lombok.ToString;
+
+import java.time.Duration;
+
+@SuppressWarnings("WeakerAccess")
+@NoArgsConstructor
+@Getter
+@Setter
+@ToString
+public class ServiceIdentifier {
+    public enum Connector {
+        ADMIN, APPLICATION
+    }
+
+    private String serviceName;
+    private String preferredVersion;
+    private String minimumVersion;
+
+    private Connector connector = Connector.APPLICATION;
+    private Duration connectTimeout = RegistryAwareClientConstants.DEFAULT_CONNECT_TIMEOUT;
+    private Duration readTimeout = RegistryAwareClientConstants.DEFAULT_CONNECT_REQUEST_TIMEOUT;
+
+    public ServiceIdentifier(String serviceName) {
+        this(serviceName, null, Connector.APPLICATION);
+    }
+
+    public ServiceIdentifier(String serviceName, String preferredVersion) {
+        this(serviceName, preferredVersion, Connector.APPLICATION);
+    }
+
+    public ServiceIdentifier(String serviceName, Connector connector) {
+        this(serviceName, null, connector);
+    }
+
+    public ServiceIdentifier(String serviceName, String preferredVersion, Connector connector) {
+        this.serviceName = serviceName;
+        this.preferredVersion = preferredVersion;
+        this.connector = connector;
+    }
+
+    public ServiceIdentifier(String serviceName, Duration connectTimeout, Duration readTimeout) {
+        this.serviceName = serviceName;
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+    }
+
+    public ServiceIdentifier(String serviceName, Connector connector, Duration connectTimeout, Duration readTimeout) {
+        this.serviceName = serviceName;
+        this.connector = connector;
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+    }
+
+    public ServiceIdentifier(String serviceName,
+                             String preferredVersion,
+                             Connector connector,
+                             Duration connectTimeout,
+                             Duration readTimeout) {
+        this.serviceName = serviceName;
+        this.preferredVersion = preferredVersion;
+        this.connector = connector;
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+    }
+
+    public ServiceIdentifier(String serviceName,
+                             String preferredVersion,
+                             String minimumVersion,
+                             Connector connector,
+                             Duration connectTimeout,
+                             Duration readTimeout) {
+        this.serviceName = serviceName;
+        this.preferredVersion = preferredVersion;
+        this.minimumVersion = minimumVersion;
+        this.connector = connector;
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+    }
+
+    /**
+     * Make a copy of {@code original).
+     */
+    public static ServiceIdentifier copy0f(ServiceIdentifier original) {
```

Comment:
Don't think we want the method name to have a zero in it.

Reply:
👍 It was copied from the scan